src/dual.jl

```diff
@@ -742,6 +742,15 @@ function LinearAlgebra.eigen(A::SymTridiagonal{<:Dual{Tg,T,N}}) where {Tg,T<:Rea
     Eigen(λ,Dual{Tg}.(Q, tuple.(parts...)))
 end
 
+# SpecialFunctions.logabsgamma           #
+# Derivative is not defined in DiffRules #
+#----------------------------------------#
+
+function SpecialFunctions.logabsgamma(d::Dual{T,<:Real}) where {T}
+    x = value(d)
+    y, s = SpecialFunctions.logabsgamma(x)
+    return (Dual{T}(y, SpecialFunctions.digamma(x) * partials(d)), s)
```

Comment:
Maybe this should return rather
```suggestion
    return (Dual{T}(y, SpecialFunctions.digamma(x) * partials(d)), Dual{T}(s, zero(s)))
```
to be consistent with `sign(::Dual)`?

Reply:
it doesn't really matter one way or the other does it? promotion will fix it when it matters.

Reply:
No, I don't think it matters. It felt easier for downstream computations and somewhat stronger to drop the partials completely. Only afterwards I noticed the inconsistency with `sign` and started to wonder if we should return a zero partial instead.

Reply:
Hmm.... Maybe we should be consistent with `sign` at all since it makes a difference in an example such as:
```julia
julia> using ForwardDiff

julia> f(x) = Inf * x
f (generic function with 1 method)

julia> g(x) = sign(x)
g (generic function with 1 method)

julia> g(x::ForwardDiff.Dual) = sign(ForwardDiff.value(x))
g (generic function with 2 methods)

julia> ForwardDiff.derivative(f ∘ g, 1.0)
0.0

julia> ForwardDiff.derivative(f ∘ sign, 1.0)
NaN
```

The "stronger" definition that drops the partial completely doesn't run into the NaN-safe mode issues and hence does not care about the non-finite result.

Maybe it's an argument for defining
```julia
Base.sign(x::Dual{<:Any,<:Real}) = sign(value(x))
```
rather than changing this PR though.